### PR TITLE
Changes for 2.0 (beta 4)

### PR DESCRIPTION
**Pull request by sroebert**
This pull request contains more changes to more closely match Vapor MySQL. This will also be needed to perform further updates to PostgreSQL Driver.

**Changes:**
- Moved all methods from `Database` to `Connection`, leaving `Database` to be a simple class containing all the information to connect to a database.
- Updated `PostgreSQLError` and replaced all uses of `DatabaseError` with it.
- Created method for performing multiple queries in one transaction, using a specific isolation level.
- Added option to the `listen` function to stop listening.
- Prevented the `listen` function from crashing with a fatal error. Now an error is returned instead.
- Refactored all binding code into a similar structure as Vapor MySQL.
- Implemented handling of `Date`, instead of parsing the date into a string.
- Added unit tests for the new date type parsing.

### CODE REVIEW

**Review by vzsg — APPROVED** [review]
Amazing!

**Comment by Prince2k3** [comment]
I approve!

**Review by Prince2k3 — APPROVED** [review]

File: Sources/PostgreSQL/Connection.swift [hunk 1]
```diff
@@ -26,109 +27,208 @@ public final class Connection: ConnInfoInitializable {
...
+            types.append(bind.type.oid ?? 0)
+            formats.append(bind.format.rawValue)
+            values.append(bind.bytes)
+            lengths.append(Int32(bind.length))
         }
-
-        let res: Result.Pointer = PQexecParams(
+        
+        let resultPointer: Result.Pointer = PQexecParams(
             cConnection, query,
```
Comment (natebird) [inline Sources/PostgreSQL/Connection.swift]: Just a nitpick but should `query` be on its own line here?

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ignore:
+  - Tests
```
Comment (natebird) [inline codecov.yml]: This still doesn't seem to be ignoring the `Tests` folder.

Maybe try `"Tests"` or `"Tests/**/*"`

**Review by natebird — APPROVED** [review]
This looks really good. Thanks for bringing it up to comparison with the MySQL library. That will make it much easier to maintain.

I made a few comments but nothing to hold up the merge.

File: Sources/PostgreSQL/Connection.swift (hunk 1 above)
Comment (sroebert) [reply Sources/PostgreSQL/Connection.swift]: Totally agree, fixed it.

File: codecov.yml (hunk 2 above)
Comment (sroebert) [reply codecov.yml]: It seems it also has to be in a `coverage` parent, changed that as well.

**Pull request merged**